File: testsuite/octave-modular/distance.m
function y = distance(filename)
	init_shogun;
	y=true;

	addpath('util');
	addpath('../data/distance');

	eval('globals'); % ugly hack to have vars from filename as globals
	eval(filename);

	if ~set_features('distance_')
		return;
	end
	if ~set_distance()
		return;
	end

	dm_train=max(max(abs(distance_matrix_train-distance.get_distance_matrix())));
	distance.init(feats_train, feats_test);
	dm_test=max(max(abs(distance_matrix_test-distance.get_distance_matrix())));

	data={'distance', dm_train, dm_test};
	y=check_accuracy(distance_accuracy, data);

File: tests/integration/r_modular/distance.R
distance <- function(filename) {
	source('util/get_features.R')
	source('util/get_distance.R')
	source('util/check_accuracy.R')

	feats <- get_features('distance_')
	if (typeof(feats)=='logical') {
		return(TRUE)
	}

	distance <- get_distance(feats)
	if (typeof(distance)=='logical') {
		return(TRUE)
	}

	dmatrix <- distance$get_distance_matrix()
	dm_train <- max(max(abs(distance_matrix_train-dmatrix)))

	distance$init(distance, feats[[1]], feats[[2]])
	dmatrix <- distance$get_distance_matrix()
	dm_test <- max(max(abs(distance_matrix_test-dmatrix)))

	data <- list(dm_train, dm_test)
	return(check_accuracy(distance_accuracy, 'distance', data))
}
